taichi/runtime/llvm/runtime.cpp

```diff
@@ -401,6 +405,7 @@ struct ListManager {
   i32 log2chunk_num_elements;
   i32 lock;
   i32 num_elements;
+  bool up_to_date;
```

Comment:
nit: `ListManager` is not only used by list-gen, but also inside `NodeManager`. Is it possible that we have a higher-level struct for list-gen? e.g.

```cpp
struct ListgenManager {
  ListManager list;
  bool up_to_date{false};
};
```

Reply:
Seems easy enough! Will do

Reply:
Or just inherit from it, i.e.,

```cpp
struct ListgenManager : public ListManager {
  bool up_to_date{false};
};
```

This seems even simpler